railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -249,6 +246,18 @@ def config_target_version
       def user_default_branch
         @user_default_branch ||= `git config init.defaultbranch`
       end
+
+      def git_init_command
+        return "git init" if user_default_branch.strip.present?
+
+        git_version = `git --version`[/\d+.\d+.\d+/]
```

Comment:
```suggestion
        git_version = `git --version`[/\d+\.\d+\.\d+/]
```

Reply:
We probably should, but I'm much less concerned about it... in the unlikely event it breaks, it's only a problem for us, so dealing with it at that time isn't a big worry. 🤷🏻‍♂️